.github/workflows/e2ePerformanceTests.yml

```
@@ -33,20 +35,33 @@ jobs:
 
       - name: Get pull request details
         id: getPullRequestDetails
-        uses: Expensify/App/.github/actions/javascript/getPullRequestDetails@main
+        uses: jczekalski/App/.github/actions/javascript/getPullRequestDetails@fix/15605-run-e2e-performance-tests-for-unmerged-PRs
         with:
           GITHUB_TOKEN: ${{ github.token }}
           PULL_REQUEST_NUMBER: ${{ inputs.PR_NUMBER }}
           user: ${{ github.actor }}
 
-      - name: Get merge commit sha for the pull request
+      - name: Get merge commit sha for the pull request (if merged)
+        if: ${{ fromJSON(steps.getPullRequestDetails.outputs.IS_MERGED) }}
         id: getMergeCommitSha
-        run: echo "MERGE_COMMIT_SHA=${{ steps.getPullRequestDetails.outputs.MERGE_COMMIT_SHA }}" >> "$GITHUB_OUTPUT"
+        run: |
+          echo "MERGE_COMMIT_SHA=${{ steps.getPullRequestDetails.outputs.MERGE_COMMIT_SHA }}" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+
+      - name: Get merge commit sha for the pull request (if not merged)
+        if: ${{ !fromJSON(steps.getPullRequestDetails.outputs.IS_MERGED) }}
+        run: |
+          git checkout main
+          git remote add pr_remote ${{ steps.getPullRequestDetails.outputs.REMOTE_URL }}
+          git fetch pr_remote
+          git merge --no-commit ${{ steps.getPullRequestDetails.outputs.HEAD_COMMIT_SHA }}
+          echo "MERGE_COMMIT_SHA=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
         env:
           GITHUB_TOKEN: ${{ github.token }}
 
   buildBaseline:
-    runs-on: ubuntu-20.04-64core
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-20.04-64core
```
Modified for testing purposes, revert change before merging to `main`.